source/MaterialXFormat/XmlIo.h

```diff
@@ -90,20 +90,28 @@ class MX_FORMAT_API ExceptionFileMissing : public Exception
 /// Read a Document as XML from the given character buffer.
 /// @param doc The Document into which data is read.
 /// @param buffer The character buffer from which data is read.
+/// @param searchPath An optional sequence of file paths that will be applied
+///    in order when searching for the given file and its includes.  This
+///    argument can be supplied either as a FileSearchPath, or as a standard
+///    string with paths separated by the PATH_SEPARATOR character.
 /// @param readOptions An optional pointer to an XmlReadOptions object.
 ///    If provided, then the given options will affect the behavior of the
 ///    read function.  Defaults to a null pointer.
 /// @throws ExceptionParseError if the document cannot be parsed.
-MX_FORMAT_API void readFromXmlBuffer(DocumentPtr doc, const char* buffer, const XmlReadOptions* readOptions = nullptr);
+MX_FORMAT_API void readFromXmlBuffer(DocumentPtr doc, const char* buffer, FileSearchPath searchPath = FileSearchPath(), const XmlReadOptions* readOptions = nullptr);
```

Comment:
Would it be cleaner to add searchPath as part of XmlReadOptions, simple string?

Reply:
It would make it cleaner. For now it's just for consistency with the file API  but perhaps that should also be cleaned up.
As for string usage, it's better to use a search path as it is a set of paths.

Reply:
Resolving this to keeping this separate to allow search path w/o additional read-options.